File: moduleView/src/main/java/com/sunny/module/view/xxx/XxxLayoutActivity.java
package com.sunny.module.view.xxx;

import android.content.res.ColorStateList;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.ViewPager;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.sunny.lib.common.base.BaseActivity;
import com.sunny.lib.common.router.RouterConstant;
import com.sunny.lib.ui.SunnyViewPage;
import com.sunny.module.view.R;

import org.jetbrains.annotations.Nullable;

/**
 * Created by zhangxin17 on 2020/11/19
 */
@Route(path = RouterConstant.View.PageXxx)
public class XxxLayoutActivity extends BaseActivity {

    //
    private BottomNavigationView mBottomNavigationView;

    //
    private SunnyViewPage mViewPage;
    private XxxFragmentAdapter mFragmentAdapter;
    private int mWhichFragment = XxxLayoutConstants.FragmentDetail;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.view_act_xxxlayout);

        initView();
    }

    private void initView() {

        //
        mViewPage = findViewById(R.id.view_page);
        initViewPage();

        //
        mBottomNavigationView = findViewById(R.id.bottom_navigation_view);
        initBottomNavigation();

    }

    private void initBottomNavigation() {

        mBottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {

                if (item.getItemId() == R.id.navigation_detail) {
                    mViewPage.setCurrentItem(XxxLayoutConstants.FragmentDetail);

                } else if (item.getItemId() == R.id.navigation_coordinator) {
                    mViewPage.setCurrentItem(XxxLayoutConstants.FragmentCoordinator);
                } else if (item.getItemId() == R.id.navigation_drawer) {
                    mViewPage.setCurrentItem(XxxLayoutConstants.FragmentDrawer);
                } else if (item.getItemId() == R.id.navigation_expandable) {
                    mViewPage.setCurrentItem(XxxLayoutConstants.FragmentExpandable);
                } else if (item.getItemId() == R.id.navigation_motion) {
                    mViewPage.setCurrentItem(XxxLayoutConstants.FragmentMotion);
                } else {
                    return false;
                }
                return true;
            }
        });

        mBottomNavigationView.setOnNavigationItemReselectedListener(new BottomNavigationView.OnNavigationItemReselectedListener() {
            @Override
            public void onNavigationItemReselected(@NonNull MenuItem item) {

            }
        });

        int[][] colorState = new int[][]{
                new int[]{-android.R.attr.state_checked}, // "-" 代表未选中
                new int[]{android.R.attr.state_checked}
        };

        int[] colors = new int[]{
                getResources().getColor(R.color.black_30),
                getResources().getColor(R.color.colorRed)
        };

        ColorStateList colorStateList = new ColorStateList(colorState, colors);

        mBottomNavigationView.setItemIconTintList(null);
        mBottomNavigationView.setItemTextColor(colorStateList);

        updateSelectedItemId(mWhichFragment);
    }

    private void updateSelectedItemId(int position) {
        int fragmentId;
        switch (position) {
            case XxxLayoutConstants.FragmentDetail:
                fragmentId = R.id.navigation_detail;
                break;

            case XxxLayoutConstants.FragmentDrawer:
                fragmentId = R.id.navigation_drawer;
                break;

            case XxxLayoutConstants.FragmentCoordinator:
                fragmentId = R.id.navigation_coordinator;
                break;

            case XxxLayoutConstants.FragmentMotion:
                fragmentId = R.id.navigation_motion;
                break;

            case XxxLayoutConstants.FragmentExpandable:
                fragmentId = R.id.navigation_expandable;
                break;

            default:
                fragmentId = R.id.navigation_detail;
                break;
        }
        mBottomNavigationView.setSelectedItemId(fragmentId);
    }

    private void initViewPage() {
        mFragmentAdapter = new XxxFragmentAdapter(getSupportFragmentManager());
        mViewPage.setAdapter(mFragmentAdapter);
        mViewPage.setOffscreenPageLimit(mFragmentAdapter.getCount());
        mViewPage.setCurrentItem(mWhichFragment);

        mViewPage.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                updateSelectedItemId(position);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
    }
}

File: moduleView/src/main/java/com/sunny/module/view/xxx/XxxLayoutActivity.kt
package com.sunny.module.view.xxx

import android.os.Bundle
import android.view.View
import androidx.appcompat.widget.Toolbar
import com.alibaba.android.arouter.facade.annotation.Route
import com.google.android.material.tabs.TabLayout
import com.sunny.lib.common.base.BaseActivity
import com.sunny.lib.common.router.RouterConstant
import com.sunny.lib.ui.SunnyViewPage
import com.sunny.module.view.R
import timber.log.Timber

/**
 * Created by zhangxin17 on 2020/11/19
 */
@Route(path = RouterConstant.View.PageXxx)
class XxxLayoutActivity : BaseActivity() {

    private lateinit var mTabLayout: TabLayout

    private var mViewPager: SunnyViewPage? = null
    private var mPagerAdapter: XxxFragmentAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        SystemBarTintManager.translucentStatus(this)
        setContentView(R.layout.view_act_xxxlayout)

        initView()
    }

    private fun initView() {
        mTabLayout = findViewById(R.id.tab_layout)
        mViewPager = findViewById(R.id.view_pager)

        initToolBar()
        initViewPage()

        mViewPager?.currentItem = DefPageIndex
    }

    private fun initToolBar() {
        val toolbar = findViewById<Toolbar>(R.id.tool_bar)
        toolbar.visibility = View.VISIBLE
        setSupportActionBar(toolbar)
        supportActionBar?.run {
            setDisplayShowTitleEnabled(false)
            setDisplayHomeAsUpEnabled(false)
        }

        //
        XxxPageList.forEach {
            mTabLayout.addTab(mTabLayout.newTab().setText(it.title))
        }

        mTabLayout.addOnTabSelectedListener(object :
            TabLayout.ViewPagerOnTabSelectedListener(mViewPager) {
            override fun onTabReselected(tab: TabLayout.Tab) {
                mPagerAdapter?.getXxxPage(tab.position)?.let {
                    Timber.i("onTabReselected :${it.title}")
                }
            }

            override fun onTabSelected(tab: TabLayout.Tab) {
                super.onTabSelected(tab)
                mPagerAdapter?.getXxxPage(tab.position)?.let {
                    Timber.i("onTabSelected :${it.title}")
                }
            }
        })
    }

    private fun initViewPage() {
        mPagerAdapter = XxxFragmentAdapter(supportFragmentManager)

        val pageAdapter = mPagerAdapter ?: return

        mViewPager?.adapter = mPagerAdapter
        mViewPager?.offscreenPageLimit = pageAdapter.count
        mViewPager?.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(mTabLayout))
    }
}